dipy/workflows/io.py

```diff
@@ -775,3 +778,125 @@ def run(
+                  condition is true, number2 otherwise.
+                - ``{sin,cos,tan}(float|complex) -> float|complex``: trigonometric sine,
+                  cosine or tangent.
+                - ``{arcsin,arccos,arctan}(float|complex) -> float|complex``:
+                  trigonometric inverse sine, cosine or tangent.
+                - ``arctan2(float1, float2) -> float``: trigonometric inverse tangent of
+                  float1/float2.
+                - ``{sinh,cosh,tanh}(float|complex) -> float|complex``: hyperbolic
+                  sine, cosine or tangent.
+                - ``{arcsinh,arccosh,arctanh}(float|complex) -> float|complex``:
+                  hyperbolic inverse sine, cosine or tangent.
+                - ``{log,log10,log1p}(float|complex) -> float|complex``: natural,
+                  base-10 and log(1+x) logarithms.
+                - ``{exp,expm1}(float|complex) -> float|complex``: exponential and
+                  exponential minus one.
+                - ``sqrt(float|complex) -> float|complex``: square root.
+                - ``abs(float|complex) -> float|complex``: absolute value.
+                - ``conj(complex) -> complex``: conjugate value.
+                - ``{real,imag}(complex) -> float``: real or imaginary part of complex.
+                - ``complex(float, float) -> complex``: complex from real and imaginary
+                  parts.
+                - ``contains(np.str, np.str) -> bool``: returns True for every string
+                  in op1 that contains op2.
+        input_files : variable string
+            Any number of Nifti1 files
+        dtype : string, optional
+            Data type of the resulting file.
+        out_dir : string, optional
+            Output directory
+        out_file : string, optional
+            Name of the resulting file to be saved.
+        """
+        vol_dict = {}
+        ref_affine = None
+        ref_shape = None
+        info_msg = ""
+        have_errors = False
+        for i, fname in enumerate(input_files, start=1):
+            if not os.path.isfile(fname):
+                logging.error(f"Input file {fname} does not exist.")
+                raise SystemExit()
+
+            if not (fname.endswith(".nii.gz") or fname.endswith(".nii")):
+                msg = (
+                    f"Wrong volume type: {fname}. Only Nifti files are supported"
+                    " (*.nii or *.nii.gz)."
+                )
+                logging.error(msg)
+                raise SystemExit()
+
+            data, affine = load_nifti(fname)
+            vol_dict[f"vol{i}"] = data
+            info_msg += f"{fname}:\n- vol index: {i}\n- shape: {data.shape}"
+            info_msg += f"\n- affine:\n{affine}\n"
+            if ref_affine is None:
+                ref_affine = affine
+                ref_shape = data.shape
+                continue
+
+            have_errors = (
+                have_errors
+                or not np.array_equal(ref_affine, affine)
```

Comment:
Do you want to allow for optional tolerance input here in case of affines that differ on a very small number?

Reply:
sounds like a good idea, let me update that